api/applicationserver_web.proto

```diff
@@ -31,6 +31,86 @@ message ApplicationWebhookIdentifiers {
+}
+
+// ApplicationWebhookTemplateField represents a custom field that needs to be filled by the user in order to use the template.
+// A field can be an API key, an username or password, or any custom platform specific field (such as region).
+// The fields are meant to be replaced inside the URLs and headers when the webhook is created.
+message ApplicationWebhookTemplateField {
+  // The ID of the field.
+  string id = 1 [(gogoproto.customname) = "ID", (validate.rules).string.max_len = 20];
+  // The name of the field.
+  string name = 2 [(validate.rules).string.max_len = 20];
+  // The description of the field.
+  string description = 3 [(validate.rules).string.max_len = 100];
+  // Secret decides if the field should be shown in plain-text or should stay hidden.
+  bool secret = 4;
+  // The default value of the field, if applicable.
+  string default_value = 5 [(validate.rules).string.max_len = 100];
+}
+
+message ApplicationWebhookTemplate {
+  ApplicationWebhookTemplateIdentifiers ids = 1 [(gogoproto.embed) = true, (gogoproto.nullable) = false, (validate.rules).message.required = true];
+
+  string name = 2 [(validate.rules).string.max_len = 20];
+  string description = 3 [(validate.rules).string.max_len = 100];
+  string logo_url = 4 [(gogoproto.customname) = "LogoURL", (validate.rules).string.uri = true];
+  string info_url = 5 [(gogoproto.customname) = "InfoURL", (validate.rules).string.uri = true];
+  string documentation_url = 6 [(gogoproto.customname) = "DocumentationURL", (validate.rules).string.uri = true];
+
+  string format = 7 [(validate.rules).string.max_len = 10];
+
+  repeated ApplicationWebhookTemplateField fields = 8;
+
+  // The HTTP headers used by the template. Both the key and the value can contain TemplateFields.
+  map<string,string> headers = 9;
+
+  // The base URL of the template. Can contain TemplateFields, in RFC 6570 format.
+  string base_url = 10 [(gogoproto.customname) = "BaseURL", (validate.rules).string.uri = true];
+
+  message Message {
+    // Path to append to the base URL. Can contain TemplateFields, in RFC 6570 format.
+    string path = 1;
+  }
+  Message uplink_message = 11;
+  Message join_accept = 12;
+  Message downlink_ack = 13;
+  Message downlink_nack = 14;
+  Message downlink_sent = 15;
+  Message downlink_failed = 16;
+  Message downlink_queued = 17;
+  Message location_solved = 18;
+}
+
+message ApplicationWebhookTemplates {
+  repeated ApplicationWebhookTemplate templates = 1;
+}
+
+message GetApplicationWebhookTemplateRequest {
+  ApplicationWebhookTemplateIdentifiers ids = 1 [(gogoproto.embed) = true, (gogoproto.nullable) = false, (validate.rules).message.required = true];
+  google.protobuf.FieldMask field_mask = 2 [(gogoproto.nullable) = false];
+}
+
+message ListApplicationWebhookTemplatesRequest {
+  google.protobuf.FieldMask field_mask = 1 [(gogoproto.nullable) = false];
+}
+
+service ApplicationWebhookTemplateRegistry {
```

Comment:
I don't see a reason why these RPCs shouldn't just be part of the `ApplicationWebhookRegistry` service

Reply:
I wanted to keep them separate to keep them independent (so each could be disabled without affecting the other). Merged in a4ea0327f.